qirlib/src/generation/qir/instructions.rs

```diff
@@ -101,52 +114,73 @@ pub(crate) fn emit<'ctx>(
                 generator.qis_rz_body(),
                 &[
                     generator.f64_to_f64(inst.theta),
-                    find_qubit(&inst.qubit).into(),
+                    get_qubit(&inst.qubit).into(),
                 ],
             );
         }
         Instruction::S(inst) => {
-            generator.emit_void_call(generator.qis_s_body(), &[find_qubit(&inst.qubit).into()]);
+            generator.emit_void_call(generator.qis_s_body(), &[get_qubit(&inst.qubit).into()]);
         }
         Instruction::SAdj(inst) => {
-            generator.emit_void_call(generator.qis_s_adj(), &[find_qubit(&inst.qubit).into()]);
+            generator.emit_void_call(generator.qis_s_adj(), &[get_qubit(&inst.qubit).into()]);
         }
         Instruction::T(inst) => {
-            generator.emit_void_call(generator.qis_t_body(), &[find_qubit(&inst.qubit).into()]);
+            generator.emit_void_call(generator.qis_t_body(), &[get_qubit(&inst.qubit).into()]);
         }
         Instruction::TAdj(inst) => {
-            generator.emit_void_call(generator.qis_t_adj(), &[find_qubit(&inst.qubit).into()]);
+            generator.emit_void_call(generator.qis_t_adj(), &[get_qubit(&inst.qubit).into()]);
         }
         Instruction::X(inst) => {
-            generator.emit_void_call(generator.qis_x_body(), &[find_qubit(&inst.qubit).into()]);
+            generator.emit_void_call(generator.qis_x_body(), &[get_qubit(&inst.qubit).into()]);
         }
         Instruction::Y(inst) => {
-            generator.emit_void_call(generator.qis_y_body(), &[find_qubit(&inst.qubit).into()]);
+            generator.emit_void_call(generator.qis_y_body(), &[get_qubit(&inst.qubit).into()]);
         }
         Instruction::Z(inst) => {
-            generator.emit_void_call(generator.qis_z_body(), &[find_qubit(&inst.qubit).into()]);
+            generator.emit_void_call(generator.qis_z_body(), &[get_qubit(&inst.qubit).into()]);
         }
-        Instruction::If(if_inst) => emit_if(generator, registers, qubits, entry_point, if_inst),
+        Instruction::Call(call) => emit_call(generator, qubits, results, call),
+        Instruction::If(if_) => emit_if(generator, qubits, results, entry_point, if_),
     }
 }
 
+fn emit_call<'ctx>(
+    generator: &CodeGenerator<'ctx>,
+    qubits: &HashMap<String, BasicValueEnum<'ctx>>,
+    results: &HashMap<String, Option<PointerValue<'ctx>>>,
+    call: &Call,
+) {
```

Comment:
Similarly here, it may be good to return `Result<()>` so that panics based on user input don't propagate out to Python and lead to confusing error messages.

Reply:
Added a todo linking to #31.